### PR TITLE
Rebranding, part deux

### PR DESCRIPTION
**Pull request by voidanix**
- cmake: nix: Update icon packaging
- cmake: nix: Add 256 and 512 icon variants
- install: win: Remove remnants of NSIS installer
- install: win: Move Windows-specific files from src
- scripts: Add icon generation script
- install: nix: Update icons
- install: win: Use BN ICO
- data: Update textures submodule

### CODE REVIEW

**Comment by voidanix** [comment]
Hmm, seems to conflict with https://github.com/blue-nebula/base/pull/262: marking as WIP.

**Comment by TheAssassin** [comment]
Please feel free to ignore #262, we'll fix that one after the icons have been fixed.

File: src/scripts/update-icons [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (TheAssassin) [inline src/scripts/update-icons:1]: This file contains bashisms, so better request bash explicitly.

```suggestion
#! /bin/bash
```

File: src/scripts/update-icons [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#
+# This script updates the application icons define in install/nix/blue_nebula_x**.png from the original-sized icon.png using ImageMagick
+# The sizes we want that are used in the for loop are also defined in install_nix.cmake
+
+for i in 16 32 48 64 128 256 512; do
+    echo "Scaling icon to ${i}x${i}..."
+    convert -resize ${i}x${i} $1 ../install/nix/blue-nebula_x${i}.png
```
Comment (TheAssassin) [inline src/scripts/update-icons:8]:
```suggestion
    convert -resize "${i}x${i}" "$1" "../install/nix/blue-nebula_x${i}.png"
```

    
Highly recommend using shellcheck to validate escaping.

File: src/scripts/update-icons [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    if [ "$i" -eq 512 ]; then
+        cp ../install/nix/blue-nebula_x${i}.png ../../data/textures/icon.png
```
Comment (TheAssassin) [inline src/scripts/update-icons:10]:
```suggestion
        cp "../install/nix/blue-nebula_x${i}.png" ../../data/textures/icon.png
```

File: src/scripts/update-icons
Comment (TheAssassin) [inline src/scripts/update-icons:1]: Consider adding a `.sh` suffix.

File: src/install/win/blue-nebula.rc
Comment (TheAssassin) [inline src/install/win/blue-nebula.rc:1]: I think the current purpose of this `src/install` directory is to house files needed to generate the installer. The `.rc` file is required during compilation to include icons into the file. However, the `.ico` is again in `src/install/win`, so there is no clear pattern. We might want to reorganize files, and possibly rename `install` to `platform` or `native` or something like that.

File: src/cmake/install_nix.cmake [hunk 4]
```diff
@@ -13,14 +13,14 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
...
             COPY ${CMAKE_CURRENT_SOURCE_DIR}/install/nix/blue-nebula_x${res}.png
-            DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}/apps/
+            DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}x${res}/apps/
         )
         file(RENAME
-            ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}/apps/blue-nebula_x${res}.png
-            ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}/apps/blue-nebula.png
+            ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}x${res}/apps/blue-nebula_x${res}.png
+            ${CMAKE_CURRENT_BINARY_DIR}/icons/hicolor/${res}x${res}/apps/blue-nebula.png
             )
```
Comment (TheAssassin) [inline src/cmake/install_nix.cmake:24]: While we're at it, mind to fix the code style?

```suggestion
        )
```

**Review by TheAssassin — CHANGES_REQUESTED** [review]

File: src/scripts/update-icons (hunk 1 above)
Comment (voidanix) [reply src/scripts/update-icons:1]: Where does it do so? It runs perfectly fine with dash, plus both shellcheck and checkbashisms do not say anything.

File: src/scripts/update-icons
Comment (voidanix) [reply src/scripts/update-icons:1]: I suppose this must be done for almost every file in scripts though?

File: src/scripts/update-icons (hunk 1 above)
Comment (TheAssassin) [reply src/scripts/update-icons:1]: Seems like I was wrong, I updated shellcheck. In any case, I think using bash makes sense, as it allows you to `set -euo pipefail`.

File: src/scripts/update-icons
Comment (TheAssassin) [reply src/scripts/update-icons:1]: Well, in the long term, I think that would be a good idea. Feel free to ignore my suggestion for now.

**Review by TheAssassin — APPROVED** [review]

**Comment by TheAssassin** [comment]
Thanks!

**Pull request merged**